### PR TITLE
Add SonarQube integration blog post

### PR DESCRIPTION
**Pull request by ceddlyburge**
Hi There

As discussed here is a blog post about integration with SonarQube.

I've also made a couple of tiny changes to the last post.

There are a couple of things that I am not 100% sure on.

Can you check that you are happy with this statement:
> Normally AppVeyor will not decrypt secure variables in Pull Requests, as in this case a Hacker could send you a PR and then read all of your secure data. However, for SonarQube to analyse Pull Requests, it is necessary. You need to decide whether you can live with this.

Can you also check that my method of creating an alternative pull request build is as you would recommend:

```yml
build_script:
 - ps: if ($env:APPVEYOR_PULL_REQUEST_NUMBER) { blah }
 - ps: if (-Not $env:APPVEYOR_PULL_REQUEST_NUMBER) { other blah }
```

Thanks and Happy Christmas :) Hopefully you are at home having fun now instead of looking at this PR. That's certainly what I am about to do!

Cedd

### CODE REVIEW

**Comment by ceddlyburge** [comment]
I notice that the htmllinter is complaining, but to fix it I think I would have to change the names of the headings, which I think would be a bad thing.

I also notice that this problem already exists for some existing posts, so I'm not going to worry about it for now.

Please let me know if you want me to change the headings to keep the linter happy.

**Comment by FeodorFitsner** [comment]
Ha, interesting! Indeed your article has two similar sections (`Create a new repository on GitHub` and `Conclusion`) as [this one](https://www.appveyor.com/blog/2016/11/22/create-signed-aligned-xamarin-apk/) and when Jekyll's doing a new blog index page it complains about duplicate headings as both posts goes to that index page. Looks like a bug in Jekyll? For workaround, give your sections slightly different names, like `Create a new GitHub repository` and `Wrapping up`.

File: src/_posts/2016-12-23-sonarqube.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+title: SonarQube Analysis
+---
+This is a guest post by [Cedd Burge](https://github.com/ceddlyburge), Software Developer Lead at [RES](http://resgroup.github.io/).
```
Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: Add a new line here for conistency.

File: src/_posts/2016-12-23-sonarqube.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+### Install chocolatey
+
+* [Install Chocolatey](https://chocolatey.org/install) from an administrator command prompt / powershell.
+* Close the command prompt
+
+### Install SonarQube MSBuild Runner
+
+* Open a new administrator command prompt / powershell.
+* ```choco install "msbuild-sonarqube-runner" -y```
```
Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: If you need inline code use one backtick otherwise break this.

File: src/_posts/2016-12-23-sonarqube.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+### Analyse and upload to SonarQube
+
+* `MSBuild.SonarQube.Runner.exe begin /k:"**YourUniqueProjectName**" /d:"sonar.host.url=https://sonarqube.com" /d:"sonar.login=**YourSonarQubeToken**"`
```
Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: Better use a code block

Comment (ceddlyburge) [inline src/_posts/2016-12-23-sonarqube.md]: Hi, I've done the first two items, but I'm less keen on this one. I did have it as a code block 
originally but it was hard to read, so I changed it to single line bullet points and it scanned a lot better. 

If you insist I'll do it of course, but I'm not keen.

Cheers

Cedd

File: src/_posts/2014-02-19-appveyor-20-dedicated-build-vms-parallel-testing-nuget-deployment.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ deployment.
 * [NuGet hosting](#nuget)
 * [Build matrix](#build-matrix)
 * [Parallel testing](#parallel-testing)
-* [Deployment to multiple environments](#deployment)
+* [Deployment to multiple environments](#deploying)
```
Comment (XhmikosR) [inline src/_posts/2014-02-19-appveyor-20-dedicated-build-vms-parallel-testing-nuget-deployment.md]: Why this change? I mean, it will break old references which we can't control ourselves.

File: src/_posts/2016-12-23-sonarqube.md (hunk 3 above)
Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: Nah, no worries. This way it looks more like steps if you want that.

File: src/_posts/2016-12-23-sonarqube.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+* Open a new administrator command prompt / powershell.
+* `choco install "msbuild-sonarqube-runner" -y`
+
+### Analyse and upload to SonarQube
+
+* `MSBuild.SonarQube.Runner.exe begin /k:"**YourUniqueProjectName**" /d:"sonar.host.url=https://sonarqube.com" /d:"sonar.login=**YourSonarQubeToken**"`
+* `"**YourPathToMSBuild**\MSBuild.exe" "**YourProjectName**.sln"`
+* `MSBuild.SonarQube.Runner.exe end /d:"sonar.login=**YourSonarQubeToken**"`
+
+When finished, you will be able to see the results at [sonarqube.com/](https://sonarqube.com/). If it isn't working, make sure you are using MSBuild 14 and [Java 1.8 or later](http://stackoverflow.com/questions/40249947/msbuild-sonarqube-runner-exe-cant-access-https-sonarqube-com). The SonarQube [Getting Started](https://about.sonarqube.com/get-started/) page is excellent if these instructions become out of date.
```
Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: `[sonarqube.com/](https://sonarqube.com)`

**Comment by XhmikosR** [comment]
Sorry for the so many comments, I think those were the last ones. You could also squash all commits into one when you are done.

File: src/_posts/2014-02-19-appveyor-20-dedicated-build-vms-parallel-testing-nuget-deployment.md (hunk 4 above)
Comment (ceddlyburge) [inline src/_posts/2014-02-19-appveyor-20-dedicated-build-vms-parallel-testing-nuget-deployment.md]: There are two posts with a "Deployment" header in, that are now on the same page (page 6). Presumably they were in different pages before.

Jekyll uses the header text in html name attributes, so they must be unique on the page.

The html linter complained about this so I changed one heading from "Deployment" to "Deploying".

Then the link checker complained so I changed the line above to fix it.

Cheers

Cedd

Comment (XhmikosR) [inline src/_posts/2014-02-19-appveyor-20-dedicated-build-vms-parallel-testing-nuget-deployment.md]: I see now. It's a bug with the paginator plugin, I believe, since we have the same issue with /updates/.

There, I just ignore it from the html linter. Perhaps I should do the same here.

For reference https://github.com/appveyor/website/blob/master/Gruntfile.js#L270

File: src/_posts/2016-12-23-sonarqube.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+Again, you can check the results at [sonarqube.com](https://sonarqube.com/).
+
+## Add a SonarQube badge to the repo
+
+There are are variety of [Quality Gate](https://github.com/QualInsight/qualinsight-plugins-sonarqube-badges/wiki/Quality-Gate-status-badges) and [Metrics](https://github.com/QualInsight/qualinsight-plugins-sonarqube-badges/wiki/Measure-badges) badges available.
+
+To add a standard [![Quality Gate](https://sonarqube.com/api/badges/gate?key=SonarQubeNemoOnAppveyor)](https://sonarqube.com/dashboard/index/SonarQubeNemoOnAppveyor) badge, add the following to readme.md.
+
+`[![Quality Gate](https://sonarqube.com/api/badges/gate?key=YourUniqueProjectName)](https://sonarqube.com/dashboard/index/YourUniqueProjectName)`
```
Comment (ceddlyburge) [inline src/_posts/2016-12-23-sonarqube.md]: Again I could do this but I would rather not. the code block doesn't wrap and its annoying not being able to see the whole thing at once.

Comment (XhmikosR) [inline src/_posts/2016-12-23-sonarqube.md]: Yeah, but that is a feature. Just use this since it will be highlighted properly.

**Comment by ceddlyburge** [comment]
No worries about the comments.

I've addressed all the problems. But there are 2 outstanding things:

1. Let me know what you want to do about the "Deployment" / "Deploying" header and link change in the 2014 post.
2. Decide whether you are happy using the single line code markup as opposed to the code blocks. In my opinion it does read a lot better in the 2 cases that you mention.

I think "Squash and Merge" is an option that you have when merging, rather than one I have. I could probably work out the git magic required to do it but I don't think its time well spent.

Cheers

Cedd

**Comment by XhmikosR** [comment]
It's just a single command `git rebase -i upstream/master` but I guess @FeodorFitsner can choose to squash when merging.

For 2 I prefer code blocks personally because we can highlight the code and it won't overflow on mobile.
For the first issue, what you did is just a workaround so since we can't fix the paginator plugin, we can either ignore the file or you choose a different name for the header.

**Comment by ceddlyburge** [comment]
1. I don't know what you want to do about the paginator problem. If I leave things as they are, then the html linter will fail. Renaming the headings is the workaround that Feodor recommended. If you prefer a different workaround please let me know what it is.

2. As mentioned, I prefer the inline code in those places, as it means it is readable without any extra sideways scrolling by the user. This is even more relevant on mobile in my opinion, when it is more fiddly to scroll the code block sideways with a fat finger. However, if you insist, I'll change it. Please let me know if you do insist.

Cheers

Cedd

**Comment by XhmikosR** [comment]
1. Leave it as is and I'll check out your branch later and fix it myself.
2. Yeah, use code blocks and specify the lang where applicable.

**Comment by XhmikosR** [comment]
@ceddlyburge: please check the commits now; I forced push your branch after cleaning things up.

**Comment by XhmikosR** [comment]
@FeodorFitsner: please merge, do **NOT** squash.

**Comment by FeodorFitsner** [comment]
Thank you, guys!
